### PR TITLE
Move type declaration to constant instead of constructor parameter.

### PR DESCRIPTION
**Pull request by mike4git**
What are you thinking about that declaration?!
Type parameter could be skipped and be consistent everywhere (by using right EditableItem type/class automatically)

### CODE REVIEW

**Comment by mike4git** [comment]
Be aware this PR is based upon a not already merged PR.

File: src/EditableDialogBox/EditableItem.php [hunk 1]
```diff
@@ -5,14 +5,16 @@
 
 class EditableItem extends DialogBoxItem
 {
+    public const TYPE = 'editable';
```
Comment (jdreesen) [inline src/EditableDialogBox/EditableItem.php:8]: This is wrong. There is no "editable" type in Pimcore.

File: src/EditableDialogBox/EditableItem.php [hunk 2]
```diff
@@ -5,14 +5,16 @@
...
     private string $name;
     private string $label = '';
     /** @var array<string, bool|float|int|string> */
     private array $config = [];
 
-    public function __construct(string $type, string $name)
+    public function __construct(string $name)
     {
-        parent::__construct($type);
+        parent::__construct();
```
Comment (jdreesen) [inline src/EditableDialogBox/EditableItem.php:17]: I don't see any advantage in having to call a constructor without arguments that will set a value based on a constant that you have to override. For me, this is too much "magic".

**Review by jdreesen — COMMENTED** [review]
I don't like this because it's not as explicit as having to define the type in the constructor. All items *need* a type, but with this `const` solution it's easy to forget to override the default `TYPE` constant, which is empty, resulting in an invalid item.

Apart from that, it would make the dynamic usage of the `EditableItem` impossible, since right now you could do this 
```php
(new EditableItem('some type not yet implemented'))->addConfig(...)
```
which would then no longer be supported.

**Pull request closed without merging**